### PR TITLE
 Use Source Han Sans for preview images generation (bis)

### PR DESCRIPTION
**Pull request by cincodenada**
This is a partial fix for #1828 - these are the same changes as #1848, I've just rewritten the history to not include the font file in the repo itself, and added a line to download it into the Docker file at build time instead.

I ran a test clone to make sure I got this branch properly scrubbed, and it looks good - I pulled this branch down and the size of my `.git` dir stayed the same as it was before (21MB), and then pulled down the original branch and (after much downloading) my `.git` dir was 42MB, so we should be good!

### CODE REVIEW

**Comment by cincodenada** [comment]
I'm still working on confirming this works, my dev machine is being annoying tonight so I'll probably have to dig back into this tomorrow. But I think this should do the trick.

**Comment by joachimesque** [comment]
Thank you! I'll close #1848

**Comment by cincodenada** [comment]
Okay, this ended up being trickier than I had anticipated - because we're mounting the app as a volume, everything under `/app` is replaced, which means we can't stash things there.

We could load the font under some other directory in the docker image (say, `/resources`), but then it's not in the `/app/static/fonts` directory, which is kinda messy. We could copy it into the `/app` directory at startup or somewhere, but at that point I figured we might as well just download it at that point instead.

So, I ended up here, where we download the font the first time we need it. Upside here is we only download it if we actually need it - so folks without the environment variable set don't need to download it. Gotchas here are race conditions - it's possible we could try to download the file twice if two parts of the code ask for the font at the same time, but I'm not sure how likely that is.

It may make more sense to just download the font at app startup instead - maybe `apps.py` would be a reasonable place to do that, as we did with celerywyrm in #1751?

So, in short: this works (altho has some tests/linting to fix, it looks like) but I'm not convinced it's the best solution here.

File: bookwyrm/settings.py [hunk 1]
```diff
@@ -310,13 +327,8 @@
     MEDIA_FULL_URL = MEDIA_URL
     STATIC_FULL_URL = STATIC_URL
     DEFAULT_FILE_STORAGE = "bookwyrm.storage_backends.ImagesStorage"
-    # I don't know if it's used, but the site crashes without it
```
Comment (mouse-reeve) [inline bookwyrm/settings.py:313]: thank you for this

Comment (cincodenada) [reply bookwyrm/settings.py:313]: It was handy, I was like "dang, I want these higher up in the file" and then realized that they were the same whether we were using S3 or not, so they could be! 😁

**Comment by mouse-reeve** [comment]
I'm trying to test this, but my images that use web are all crashing becuase they can't find the font file. The web image itself give me the message `Just a sec, downloading Source Han Sans`, but still crashes with the file not found error.

**Comment by cincodenada** [comment]
Hmm! Thanks for testing, I'll try spinning up a new instance myself when I have a minute to see if I can reproduce. Can you paste the error trace you're getting? If it can't find the file it should use the fallback font rather than crashing, so that's extra strange!

**Comment by mouse-reeve** [comment]
This is the error I'm seeing:
```
web_1             | Django version 3.2.10, using settings 'bookwyrm.settings'
web_1             | Starting development server at http://0.0.0.0:8000/
web_1             | Quit the server with CONTROL-C.
web_1             | /app/bookwyrm/settings.py changed, reloading.
web_1             | Just a sec, downloading Source Han Sans
web_1             | Watching for file changes with StatReloader
web_1             | Exception in thread django-main-thread:
web_1             | Traceback (most recent call last):
web_1             |   File "/usr/local/lib/python3.9/threading.py", line 950, in _bootstrap_inner
web_1             |     self.run()
web_1             |   File "/usr/local/lib/python3.9/threading.py", line 888, in run
web_1             |     self._target(*self._args, **self._kwargs)
web_1             |   File "/usr/local/lib/python3.9/site-packages/django/utils/autoreload.py", line 64, in wrapper
web_1             |     fn(*args, **kwargs)
web_1             |   File "/usr/local/lib/python3.9/site-packages/django/core/management/commands/runserver.py", line 110, in inner_run
web_1             |     autoreload.raise_last_exception()
web_1             |   File "/usr/local/lib/python3.9/site-packages/django/utils/autoreload.py", line 87, in raise_last_exception
web_1             |     raise _exception[1]
web_1             |   File "/usr/local/lib/python3.9/site-packages/django/core/management/__init__.py", line 375, in execute
web_1             |     autoreload.check_errors(django.setup)()
web_1             |   File "/usr/local/lib/python3.9/site-packages/django/utils/autoreload.py", line 64, in wrapper
web_1             |     fn(*args, **kwargs)
web_1             |   File "/usr/local/lib/python3.9/site-packages/django/__init__.py", line 24, in setup
web_1             |     apps.populate(settings.INSTALLED_APPS)
web_1             |   File "/usr/local/lib/python3.9/site-packages/django/apps/registry.py", line 122, in populate
web_1             |     app_config.ready()
web_1             |   File "/app/bookwyrm/apps.py", line 40, in ready
web_1             |     download_file(config["url"], font_path)
web_1             |   File "/app/bookwyrm/apps.py", line 17, in download_file
web_1             |     with open(destination, "b+w") as outfile:
web_1             | FileNotFoundError: [Errno 2] No such file or directory: '/app/static/fonts/source_han_sans/SourceHanSans-VF.ttf.ttc'
```

**Comment by cincodenada** [comment]
Okay, great, that makes sense. It's a confusing error: the actual error here is that the directory `/app/static/fonts/source_han_sans` doesn't exist, so it can't create the fontfile to write to it.

Which...after looking closer at how we build up the docker things, I see the problem, I think - I forgot (again) that the static directories aren't mounted like the rest of the app, so creating the directory in the repo doesn't actually create it in the image.

It _might_ work if you run `collectstatic` and then try again, but that's just a workaround - we should be creating the directory here (and catching file errors) anyway. I'll update the PR today to do that. Thanks!

**Comment by cincodenada** [comment]
Okay, just pushed up a commit that should create the directory, and also not crash if we can't, I tested both of those by meddling around with my local dockers, and they both worked as expected. Thanks for testing!
